docs/src/nondiff_points.md

```diff
@@ -0,0 +1,141 @@
+# What to return for non-differentiable points
+!!! info "What is the short version?"
+    If the function is not differentiable due to e.g. a branch, like `abs`, your rule can reasonably claim the derivative at that point is the value from either branch, *or* any value in-between (e.g. for `abs` claiming 0 is a good idea).
+    If it is not differentiable due to the primal not being defined on one side, you can set it to what ever you like.
+    Your rule should claim a derivative that is *useful*.
+In calculus one learns that if the derivative as computed by approaching from the left,
+and the derivative one computes as approaching from the right are not equal then the derivative is not defined,
+and we say the function is not differentiable at that point.
+This is distinct from the notion captured by [`NoTangent`](@ref), which is that the tangent space itself is not defined: because in some sense the primal value can not be perturbed e.g. is is a discrete type.
+
+However, contrary to what calculus says most autodiff systems will return an answer for such functions.
+For example for: `abs_left(x) = (x <= 0) ? -x : x`, AD will say the derivative at `x=0` is `-1`.
+Alternatively for:  `abs_right(x) = (x < 0) ? -x : x`, AD will say the derivative at `x=0` is `1`.
+Those two examples are weird since they are equal at all points, but AD claims different derivatives at `x=0`.
+The way to fix autodiff systems being weird is to write custom rules.
+So what rule should we write for this case?
+
+The obvious answer, would be to write a rule that throws an error if input at a point where calculus says the derivative is not defined.
+Another option is to return some error signally value like `NaN`.
+Which you *can* do.
+However, there is no where to go with an error, the user still wants a derivative; so this is not useful.
+
+Let us explore what is useful:
+## Case Studies
+
+```@setup nondiff
+using Plots
+gr(framestyle=:origin, legend=false)
+```
+### Derivative is defined in usual sense
+```@example nondiff
+plot(x->x^3)
+```
+This is the standard case, one can returned the derivative that is defined according to school room calculus.
+Here we would reasonably say that at `x=0` the derivative is `3*0^2=0`. 
+
+
+An interesting thing about `x->x^3` is that at `x=0` the derivative is defined, but neither the sub-derivative nor super-derivative is defined.
```

Comment:
Should this line be here given that we're no longer going to talk about sub/super ?

Reply:
```suggestion
```